streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -922,6 +1030,60 @@ private InternalTopicConfig createInternalTopicConfig(final StateStoreSupplier s
     }
 
     /**
+     * Get the Pattern to match all topics requiring to start reading from earliest available offset
+     * @return the Pattern for matching all topics reading from earliest offset, never null
+     */
+    public synchronized Pattern earliestResetTopicsPattern() {
```

Comment:
About this pattern matching thing. It is unclear to me, if this is safe or not (I believe it is safe, but I want to make sure, we do not introduce a bug with it).

If a user specifies multiple input streams, and for some uses pattern and for other uses regular string, we need to ensure that those topic do not overlap (this is checked already in the current code). So I am not sure, if we need to consider this here, too? Also make sure, that for the generated pattern, there is no "overlap" for all streams using "latest" and all streams using "earliest", ie, both patterns don't match overlapping topics.

@guozhangwang WDYT?